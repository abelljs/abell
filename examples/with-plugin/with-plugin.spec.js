const fs = require('fs');
const path = require('path');

const {
  preTestSetup,
  getSelector
} = require('../../tests/test-utils/helpers.js');

describe('examples/with-plugin', () => {
  const $ = {};
  beforeAll(async () => {
    await preTestSetup('with-plugin');
    $.index = getSelector(path.join(__dirname, 'dist', 'index.html'));
    $.content = getSelector(
      path.join(__dirname, 'dist', 'hello-custom-blog', 'index.html')
    );
  });

  describe('beforeBuild plugin', () => {
    it('should have a hello-custom-blog folder', () => {
      expect(
        fs.existsSync(path.join(__dirname, 'dist', 'hello-custom-blog'))
      ).toBe(true);
    });

    it('should render h1 of blog from content into the output html', () => {
      expect($.content('[data-test="blog-content"] > #hello').html()).toBe(
        'Hello'
      );
    });
  });

  describe('beforeHTMLWrite plugin', () => {
    it('should add additional css from beforeWriteHTML plugin', () => {
      expect(
        $.index('head > style')
          .html()
          .replace(/\n|\r|\s/g, '')
      ).toBe('body{background-color:green;}');
    });

    it('should render all the slugs of blog', () => {
      expect($.index('[data-test="all-slugs"]').html()).toMatchSnapshot();
    });
  });

  describe('afterBuild plugin', () => {
    it('creates after-build.txt file', () => {
      expect(
        fs.existsSync(path.join(__dirname, 'dist', 'after-build.txt'))
      ).toBe(true);
    });
  });
});
