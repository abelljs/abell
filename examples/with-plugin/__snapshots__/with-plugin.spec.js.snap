// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`examples/with-plugin beforeHTMLWrite plugin should render all the slugs of blog 1`] = `
"
      <div>
            <a href=\\"my-first-blog\\">My First Blog</a><br>
            <span class=\\"meta-slug\\">my-first-blog</span>
          </div>
          
          <div>
            <a href=\\"another-blog\\">Another blog</a><br>
            <span class=\\"meta-slug\\">another-blog</span>
          </div>
          
          <div>
            <a href=\\"blog-from-plugin\\">blog-from-plugin</a><br>
            <span class=\\"meta-slug\\">blog-from-plugin</span>
          </div>
          
          <div>
            <a href=\\"hello-custom-blog\\">hello-custom-blog</a><br>
            <span class=\\"meta-slug\\">hello-custom-blog</span>
          </div>
      "
`;
